app/services/user_event_creator.rb

```diff
@@ -66,15 +91,21 @@ def assign_device_cookie(device_cookie)
     cookies.permanent[:device] = device_cookie unless device_cookie == cookies[:device]
   end
 
-  def send_new_device_notification(user:, device:, event:)
-    disavowal_token = EventDisavowal::GenerateDisavowalToken.new(event).call
+  def send_new_device_notification(user:, device:, disavowal_token:)
     UserAlerts::AlertUserAboutNewDevice.call(user, device, disavowal_token)
   end
 
-  def create_event_for_device(event_type:, user:, device:)
-    Event.create(
+  def create_event_for_device(event_type:, user:, device:, disavowal_token: nil)
+    disavowal_token_fingerprint = if disavowal_token
+                                    Pii::Fingerprinter.fingerprint(disavowal_token)
+                                  end
+    event = Event.create(
       user_id: user.id, device_id: device&.id, ip: request&.remote_ip, event_type: event_type,
+      disavowal_token_fingerprint: disavowal_token_fingerprint
     )
+
+    event.disavowal_token = disavowal_token
+    event
```

Comment:
maybe a bigger refactor than is good for this PR, but what if we removed the transient `disavowal_token` plaintext attribute and just returned it separately? ex

```suggestion
    [event, disavowal_token]
```

Reply:
It's also kind of a mess and could probably stand to be refactored.

Reply:
Had some bandwidth, gave it a shot in https://github.com/18F/identity-idp/pull/7305